### PR TITLE
feat(demos): redesign ChESS and Delaunay demos per design handoff

### PR DESCRIPTION
**Pull request by VitalyVorobyev**
## Summary

Implements the four-layout redesign described in `docs/design_handoff_demo_layouts/README.md` (Layout A + mobile M1 for ChESS, Layout D + mobile M1 for Delaunay). Builds on top of #62 — please merge that first or rebase this branch after.

The handoff committed under `docs/` is included so reviewers have the full design context inline.

## What changed, in three commits

1. **Primitives + camera-pose math** — `src/components/illustrations/_shared/primitives.tsx` exports theme-aware `Panel`, `PanelFlat`, `FloatingPanel`, `TinyBrow`, `Eyebrow`, `MetricCell`, `Kbd`, `Pill`, `Note`. Two new tokens (`--border-strong`, `--surface-2`) added to both light and dark themes. `cameraPose.ts` does a closed-form homography → pose decomposition (3×3 polar decomposition, no math library) with a `aspect` parameter so a frontal grid yields ~0 reproj error. 7 unit tests.
2. **Delaunay redesign (Layout D + mobile M1)** — canvas-first spatial editor with floating tool palette (Add / Move / Delete / Grid / Hover, plus Random / Clear), grid-config popover anchored to the Grid tool, top-center layer chips, top-right live coord readout, bottom-right Stats + Camera-pose HUD with mini horizon viz, bottom-left hover tooltip, kbd shortcuts (A / V / G / H / R / Delete). Mobile: pose chip + stats one-liner over the canvas, 5-button bottom dock, layer pill row, pose modal sheet, More sheet for grid steppers / random N / clear. Engine extended with `activeTool`, `pointer`, `gridPopoverOpen`, `poseModalOpen`. `DelaunayVoronoiCanvas` is now tool-aware. Drag clamping, finite filter, NumberInput wheel are preserved.
3. **ChESS redesign (Layout A + mobile M1)** — hero canvas Panel with floating Inspector top-right (Rotation / Blur / Contrast sliders + 2×2 overlay toggle grid, Pin and collapse buttons), live response HUD bottom-left (SR / DR / MR / R + formula), centered bottom dock (pattern segmented + Play/Pause + θ scrub). Mobile: stacked layout with status pills, 4-column metric strip, slider card (18 px touch thumbs), full-width Play button, collapsible Overlays. `ChessResponseSvg` / `useChessResponse` / `useChessResponseAnimation` reused unchanged. `ChessResponseIllustration` (article-page consumer) inlines the prior controls so its public API is preserved.

## Resolved open questions (from the handoff)

| Question | Choice |
|---|---|
| PR scoping | Single PR, three commits |
| Pose intrinsics | Assumed (`f = canvasWidth`, principal point centred), no controls. Source aspect = `cols/rows` |
| Theming | Theme-aware via existing tokens. Floating chrome works in both light and dark mode |

## Test plan

- [x] `bun run lint` clean
- [x] `npx vitest run` — 192 / 192 (7 new pose tests)
- [x] `bun run build` — green
- [x] Playwright walkthrough at desktop + mobile widths for both demos: tool palette, layer chips, grid popover, corner drag updates pose, layer toggles, hover tooltip, delete tool, mobile pose modal, mobile More sheet
- [x] Article illustration (`ChessResponseIllustration`) rendering preserved (consumed by `useArticleIllustrations.tsx`)

## Notes for the reviewer

- Both desktop and mobile layouts are mounted in the DOM at all times and toggled via `display: none`. Slight extra paint cost; can revisit with a media-query hook if it bites.
- `--border-strong` / `--surface-2` are new shared tokens — keep them in mind if you add more dark/light theme variants.
- Pose readout reproj error is non-zero when grid corners don't match `cols/rows` aspect — that's intentional and serves as a "your corners are skewed" signal.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/components/illustrations/delaunay-voronoi/useDelaunayVoronoi.ts [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+            ) as [Point, Point, Point, Point];
+            return { ...state, grid: { ...state.grid, corners: newCorners } };
+        }
+        case "RESET_GRID":
+            return { ...state, grid: { ...state.grid, corners: defaultCorners() } };
+        case "CLEAR_POINTS":
+            return { ...state, points: [], selectedId: null, hover: null };
+        case "RANDOM_POINTS": {
+            const padding = 40;
+            const newPoints: Point[] = Array.from({ length: action.count }, () => ({
```
[the remark this answers is not in the document]
Comment (VitalyVorobyev) [reply src/components/illustrations/delaunay-voronoi/useDelaunayVoronoi.ts:127]: Addressed in 7b873bd (on #64, which supersedes this PR): the `RANDOM_POINTS` reducer case clamps `count` to `[0, 2000]` (also rejects NaN/negative) before allocating. Test: `randomPoints clamps oversized counts (no runaway allocation)`.

File: src/components/illustrations/delaunay-voronoi/homography.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    for (let i = 0; i < 4; i++) {
+        const { x: sx, y: sy } = src[i];
+        const { x: dx, y: dy } = corners[i];
+        A.push([sx, sy, 1, 0, 0, 0, -dx * sx, -dx * sy]);
+        b.push(dx);
+        A.push([0, 0, 0, sx, sy, 1, -dy * sx, -dy * sy]);
+        b.push(dy);
+    }
+    const h = solve8(A, b);
+    if (!h) return [1, 0, 0, 0, 1, 0, 0, 0, 1];
```
[the remark this answers is not in the document]
Comment (VitalyVorobyev) [reply src/components/illustrations/delaunay-voronoi/homography.ts:46]: Addressed in 7b873bd (on #64): `computeHomography` returns `null` on singular systems; `projectGridPoints` skips projection (preserves user overrides). New tests cover the coincident- and collinear-corner cases.

File: src/components/illustrations/delaunay-voronoi/DelaunayVoronoiDesktopD.tsx [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+                        <TinyBrow className="mb-1.5">Hover</TinyBrow>
+                        <div className="flex justify-between text-xs">
+                            <span className="text-muted-foreground capitalize">{hover.kind} #{hover.index}</span>
+                            <span className="font-mono">area {hover.area.toFixed(4)}</span>
+                        </div>
+                        {hover.kind === "triangle" && (
+                            <div className="flex justify-between text-xs mt-1">
+                                <span className="text-muted-foreground">min angle</span>
+                                <span className={`font-mono ${minAngleTone(minAngleDeg) === "good" ? "text-emerald-400" : minAngleTone(minAngleDeg) === "warn" ? "text-amber-400" : "text-rose-400"}`}>
+                                    {minAngleDeg > 0 ? `${minAngleDeg.toFixed(1)}°` : "—"}
```
[the remark this answers is not in the document]
Comment (VitalyVorobyev) [reply src/components/illustrations/delaunay-voronoi/DelaunayVoronoiDesktopD.tsx:239]: Addressed in dc2e4a9 (on #64): `HoverTarget` for triangles now carries its own `minAngleDeg`, computed during hover detection in `DelaunayVoronoiCanvas.tsx`. The desktop hover panel renders that per-triangle value with the matching tone.

**Comment by VitalyVorobyev** [comment]
Superseded by #64 (merged as 1ccc64a). All review feedback addressed in the squash-merged history; the redesign commits land on `main` via that merge.
